Plug-Ins/Menu Editor/MenuItem.swift

```diff
@@ -0,0 +1,207 @@
+        self.manager = manager
+        
+        super.init()
+        
+        if iconID != 0,
+           let res = manager.findResource(type: ResourceType("ICON"), id: iconID, currentDocumentOnly: false) {
+            res.preview({ img in
+                self.willChangeValue(forKey: "iconImage")
+                self.iconImage = img
+                self.didChangeValue(forKey: "iconImage")
+            })
+        }
+
+    }
+    
+}
+
+extension MenuItem {
+    
+    override func value(forKey key: String) -> Any? {
+        if key == "markCharacter" {
+            return markCharacter
+        } else if key == "keyEquivalent" {
+            return keyEquivalent
+        } else if key == "name" {
+            return name
+        } else if key == "menuID" {
+            return menuID
+        } else if key == "mdefID" {
+            return mdefID
+        } else if key == "menuCommand" {
+            if commandsSize == .int32 {
+                return String(data: Data([UInt8((menuCommand & 0xff000000) >> 24),
+                                          UInt8((menuCommand & 0x00ff0000) >> 16),
+                                          UInt8((menuCommand & 0x0000ff00) >> 8),
+                                          UInt8((menuCommand & 0x000000ff))]), encoding: .macOSRoman)
+            } else {
+                return "\(menuCommand)"
+            }
+        } else if key == "styleByte" {
+            return styleByte
+        } else if key == "iconType" {
+            return iconType
+        } else if key == "isEnabled" {
+            return isEnabled
+        } else if key == "iconID" {
+            return "\(iconID)"
+        } else if key == "submenuID" {
+            return "\(submenuID)"
+        } else if key == "iconImage" {
+            return iconImage
+        } else if key == "isItem" {
+            return isItem
+        } else if key == "hasCommand" {
+            return hasCommand
+        } else if key == "textColor" {
+            return isEnabled ? NSColor.black : NSColor.lightGray
+        } else if key == "hasKeyEquivalent" {
+            return hasKeyEquivalent
+        } else {
+            return super.value(forKey: key)
+        }
+    }
+    
+    override func setValue(_ value: Any?, forKey key: String) {
+        if key == "markCharacter" {
+            markCharacter = value as? String ?? ""
+        } else if key == "keyEquivalent" {
+            keyEquivalent = value as? String ?? ""
+        } else if key == "name" {
+            name = value as? String ?? ""
+        } else if key == "isEnabled" {
+            isEnabled = value as? Bool ?? true
+        } else if key == "iconID" {
+            iconID = Int(value as? String ?? "0") ?? 0
+        } else if key == "submenuID" {
+            submenuID = Int(value as? String ?? "0") ?? 0
+        } else if key == "menuCommand" {
+            if commandsSize == .int32 {
+                menuCommand = UInt32(value as? String ?? "\0\0\0\0")
```

Comment:
Looks like you're using the reverse extension already 👍
You can just use a blank string for 0 though
```suggestion
                menuCommand = UInt32(value as? String ?? "")
```

Reply:
NB — I only accidentally found this extension. Usually `Int(_: String)` parses a decimal number. Might be a better idea to change this to `UInt32(fourCharCode: String)` so people like me don't accidentally use it for num -> String when it's not a four character code. And maybe the same for the way back, because it's a fourCharCode, not a decimal number as a String. I'm actually surprised that doesn't have a conflict with the existing version of `UInt32(_ : String)`. I guess your extension has priority over whatever it inherits via generics.

Reply:
Yeah, definitely. I realised the error of my ways sometime after I made those but never got around to changing them 😅
I'll do that after this gets merged.